projects/igniteui-angular/src/lib/grid/grid.component.ts

```diff
@@ -1909,6 +1909,8 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
     private _columnWidth: string;
     private _columnWidthSetByUser = false;
 
+    private _defaultTargetBodyHeight = 500;
```

Comment:
So, by using a default `height` we're not sure how many records would be displayed, nor whether all records would be fully in view. Correct?

Reply:
If the records in the data view are not enough to span the whole 500px (based on their row height, based itself on the displayDensity set) the height will be their total height instead (and no vertical scrollbar will appear).

We could always do it the other way around - let's say we want Math.min(<number of records in data view>, 10)  records in the view - then we set body height equal to that * rowHeight. In this way the arbitrary number is the number of records, not the height. The caveat is that it's either 
 different based on the displayDensity or we get different default heights for each displayDensity.

Reply:
If it works with a number of records, it shouldn't care about the displayDensity. The records would just render with the display density that is set.

Reply:
Sure, let me know if you want me to change it. Have in mind the grid will still provide height to igxFor as per the directive's spec. It is just going to derive it based on an arbitrary number of records (10 to make it 500px in default density) instead of arbitrary height.

Reply:
We agreed on rendering height that allows for 10 records to be fully visible. This height will therefore differ between `displayDensity` settings. I'll make another PR with the proper changes.